ts/ui/menu/AnnotationMenu.ts

```diff
@@ -0,0 +1,155 @@
+ */
+export function copyAnnotations(cache: [string, string][]) {
+  return (menu: MJContextMenu, sub: Submenu) => {
+    const annotations = cache.slice();
+    cache.length = 0;
+    return createAnnotationMenu(menu,  sub, annotations,
+                                () => MenuUtil.copyToClipboard(annotation.trim()));
+  };
+}
+
+/**
+ * Find the top-most semantics element that encloses the contents of the expression (if any)
+ *
+ * @returns {MmlNode | null} The semantics node that was found (or null)
+ */
+function getSemanticNode(menu: MJContextMenu): MmlNode | null {
+  let node: MmlNode = menu.mathItem?.root;
+  while (node && !node.isKind('semantics'))  {
+    if (node.isToken || node.childNodes.length !== 1) return null;
+    node = node.childNodes[0];
+  }
+  return node;
+}
+
+/**
+ * Computes the annotations for the math item the context menu is opened on.
+ *
+ * @param {MmlNode} node           The semantics node whose annotations are to be obtained
+ * @param {AnnotationTypes} types The legitimate annotation types.
+ * @param {[string, string][]} annotations The annoations cache of [type, text]
+ *    where the type is the annotation type and text is the content of the
+ *    annotation of that type.
+ */
+function getAnnotation(node: MmlNode, types: AnnotationTypes, annotations: [string, string][]) {
+  if (!node) return;
+  for (const child of node.childNodes) {
+    if (child.isKind('annotation')) {
+      const match = annotationMatch(child, types);
+      if (match) {
+        const value = child.childNodes.reduce((text, chars) => text + chars.toString(), '');
+        annotations.push([match, value]);
+      }
+    }
+  }
+}
+
+/**
+ * @param {MmlNode} child The annotation node to check if its encoding is one
+ *     of the displayable ones.
+ * @returns {string | null} The annotation type if it exists, o/w null.
+ */
+function annotationMatch(child: MmlNode, types: AnnotationTypes): string | null {
+  const encoding = child.attributes.get('encoding') as string;
+  for (const type of Object.keys(types)) {
+    if (types[type].indexOf(encoding) >= 0) {
+      return type;
+    }
+  }
+  return null;
+}
+
+/**
+ * The annotation selected in the Annotation submenu (needed for the info box to
+ * be able to show it)
+ */
+export let annotation: string = '';
+
+/**
+ * Create a submenu from the available annotations and attach it to the proper menu item
+ *
+ * @param {MJContextMenu} menu The context menu for which the submenu is constructed.
+ * @param {Submenu} submenu The submenu to attach elements to.
+ * @param {[string, string][]} annotations The annotations to use for the submenu.
+ * @param {() => void} action The action to perform when the annotation is selected.
+ * @return {Submenu} The newly created submenu.
+ */
+function createAnnotationMenu(menu: MJContextMenu, submenu: Submenu,
+                              annotations: [string, string][],
+                              action: () => void): Submenu {
```

Comment:
I had to change `Submenu` to `SubMenu` as the return type (and import `SubMenu`) in order to get this branch to compile without errors.  (Your favorite: capitalization ambiguities.)

Reply:
I also had compilation problems now. Not sure why that did not happen before. Odd that.